BWT/BWT/BWT.cs

```diff
@@ -0,0 +1,127 @@
+﻿// <copyright file="BWT.cs" company="Polina Lukianova">
+// Copyright (c) Polina Lukianova. All rights reserved.
+// </copyright>
+
+namespace BWT;
+
+public class BWT
+{
+    public static (string EncodedText, int EndPosition) TextEncode(string text)
+    {
+        List<string> permutations = [];
+        for (int i = 0; i < text.Length; ++i)
+        {
+            permutations.Add(text[i..] + text[..i]);
+        }
+
+        permutations.Sort();
+
+        string encodedText = string.Empty;
+        int endPosition = 0;
+
+        for (int i = 0; i < permutations.Count; ++i)
+        {
+            encodedText += permutations[i][^1];
+            if (permutations[i] == text)
+            {
+                endPosition = i;
+            }
+        }
+
+        return (encodedText, endPosition);
+    }
+
+    public static string TextDecode(string text, int endPosition)
+    {
+        List<string> rebuildPermutations = [];
+
+        for (int i = 0; i < text.Length; ++i)
+        {
+            rebuildPermutations.Add(string.Empty);
+        }
+
+        for (int i = 0; i < rebuildPermutations.Count; ++i)
+        {
+            for (int j = 0; j < text.Length; ++j)
+            {
+                rebuildPermutations[j] = text[j] + rebuildPermutations[j];
+            }
+
+            rebuildPermutations.Sort();
+        }
+
+        return rebuildPermutations[endPosition];
+    }
+
+    public static (string EncodedText, int EndPosition) TextEncodeOptimized(string text)
+    {
+        int[] suffixes = new int[text.Length];
```

Comment:
```suggestion
        var suffixes = new int[text.Length];
```
Тут и ниже. Не пишите один тип дважды в одном объявлении